cfe_internal/enterprise/federation/federation.cf

```diff
@@ -0,0 +1,210 @@
+body file control { inputs => { "$(sys.libdir)/stdlib.cf" }; }
+
+bundle agent federation_config
+{
+  vars:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_dir" string => "/opt/cfengine/federation";
+      "bin_dir" string => "$(federation_dir)/bin";
+      "path" string => "$(federation_dir)/cfapache/federation-config.json";
+  classes:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_enabled"
+        expression => fileexists("$(path)"),
+        scope => "namespace";
+  vars:
+    federation_enabled::
+      "data" data => readjson("$(path)");
+      "remotes" slist => getindices(@(data[remote_hubs]));
+  classes:
+    federation_enabled::
+      "federation_am_feeder"
+        expression => strcmp("superhub", "$(data[remote_hubs][$(remotes)][role])"),
+        scope => "namespace";
+      "federation_am_superhub"
+        expression => strcmp("feeder", "$(data[remote_hubs][$(remotes)][role])"),
+        scope => "namespace";
+      "federation_am_transporter"
+        and => {strcmp("superhub", "$(data[remote_hubs][$(remotes)][role])"),
+                strcmp("push_over_rsync", "$(data[remote_hubs][$(remotes)][transport][mode])") },
+        scope => "namespace";
+  reports:
+    federation_enabled::
+      "Federation enabled!";
+    federation_am_superhub::
+      "I'm a superhub!";
+    federation_am_feeder::
+      "I'm a feeder!";
+    federation_am_transporter::
+      "I'm a transporter!";
+}
+
+bundle agent federation_manage_cfdrop
+{
+  vars:
+      "user"
+        string => "cfdrop";
+      "home"
+        string => "/home/$(user)";
+      "create_files"
+        slist => {
+          "$(home)/.",
+          "$(home)/.ssh/.",
+          "$(home)/.ssh/authorized_keys"
+        };
+  classes:
+      "selinux_enabled"
```

Comment:
Probably, we should guard these classes and vars promises above to the proper context (pre-eval, will cause them to get some level of evaluation any time the policy is present in inputs) to federation_am_superhub::